packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart

```diff
@@ -99,49 +94,65 @@ class _CpuProfilerState extends State<CpuProfiler>
           ],
         ),
         Expanded(
-          child: _buildCpuProfileDataView(textTheme),
+          child: _buildCpuProfileDataView(),
         ),
       ],
     );
   }
 
-  Widget _buildCpuProfileDataView(TextTheme textTheme) {
+  Widget _buildCpuProfileDataView() {
     if (widget.data != null) {
       return widget.data.isEmpty
-          ? _buildEmptyDataView(textTheme)
+          ? _buildEmptyDataView()
           : TabBarView(
               physics: const NeverScrollableScrollPhysics(),
               controller: _tabController,
               children: _buildProfilerViews(),
             );
     } else {
-      // If [data] is null, we can assume that CPU profile data is being fetched
-      // and processed.
-      return const Center(child: CircularProgressIndicator());
+      // If [data] is null, CPU profile data is either being processed or it is
```

Comment:
can we simplify the number of cases by ensuring we always specify an empty data object rather than a null data object unless we are actively processing?